### PR TITLE
[Feature] Support TTA and add `--tta` in `tools/test.py`.

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

To support Test-Time-Aug (TTA).

## Modification

- Add `AverageScoreTTAModel` inherited from the `BaseTTAModel` in MMEngine.
- Add `--tta` option in `tools/test.py` to enable TTA during the test.

## Use cases

**To use the default TTA (average the classification scores of images w/ and w/o flip)**
```bash
python tools/test.py configs/resnet/resnet50_8xb32_in1k.py checkpoints/resnet50_in1k.pth --tta
```

**To use your own custom TTA method**
1. Add `tta_model` and `tta_pipeline` in your config file:
```python
...
tta_model = dict(type='AverageScoreTTAModel')
tta_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='ResizeEdge', scale=384, edge='short'),
    dict(type='CenterCrop', crop_size=384),
    dict(type='TestTimeAug',
         transforms=[
             [
                 dict(type='Resize', scale=224),
                 dict(type='Resize', scale=384)
             ],
             [
                 dict(type='RandomFlip', prob=1.),
                 dict(type='RandomFlip', prob=0.)
             ],
             [dict(type='PackClsInputs')],
         ]),
]
```
2. test with TTA
```bash
python tools/test.py configs/resnet/resnet50_8xb32_in1k.py checkpoints/resnet50_in1k.pth --tta
```
## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: tools/test.py [hunk 1]
```diff
@@ -56,6 +56,13 @@ def parse_args():
         '--no-pin-memory',
         action='store_true',
         help='whether to disable the pin_memory option in dataloaders.')
+    parser.add_argument(
+        '--tta',
+        action='store_true',
+        help='Whether to enable the Test-Test-Aug (TTA). If the config file '
```
Comment (yingfhu) [inline tools/test.py:62]: Test-Time-Aug

**Review by yingfhu — CHANGES_REQUESTED** [review]
Please add UT

File: tools/test.py (hunk 1 above)
Comment (mzr1996) [reply tools/test.py]: Fixed.

**Review by yingfhu — APPROVED** [review]
LGTM

**Comment by mzr1996** [comment]
This PR depends on https://github.com/open-mmlab/mmengine/pull/611.

File: mmcls/models/tta/score_tta.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+from mmengine.model import BaseTTAModel
+from mmengine.structures import BaseDataElement
+
+from mmcls.registry import MODELS
+from mmcls.structures import ClsDataSample
+
+
+@MODELS.register_module()
+class AverageScoreTTAModel(BaseTTAModel):
```
Comment (Ezra-Yu) [inline mmcls/models/tta/score_tta.py:12]: I think it makes more sense to call it AverageTTAClassifier, at least the search cannot use this.

File: mmcls/models/tta/score_tta.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    def merge_preds(
+        self,
+        data_samples_list: List[List[ClsDataSample]],
+    ) -> List[BaseDataElement]:
+        merged_data_samples = []
+        for data_samples in data_samples_list:
+            merged_data_samples.append(self.merge_single_sample(data_samples))
+        return merged_data_samples
+
+    def merge_single_sample(self, data_samples):
```
Comment (Ezra-Yu) [inline mmcls/models/tta/score_tta.py:23]: If here is a public function, need docstring and name `merge_single_sample` may make people confused if the params are the name of `data_samples`

**Review by Ezra-Yu — COMMENTED** [review]
How to use custom tta needs to be reflected in the config as well as in the documentation, but this can be done later.

File: mmcls/models/tta/score_tta.py (hunk 2 above)
Comment (mzr1996) [reply mmcls/models/tta/score_tta.py]: Ok, I have renamed it to `AverageClsScoreTTA`.

File: mmcls/models/tta/score_tta.py (hunk 3 above)
Comment (mzr1996) [reply mmcls/models/tta/score_tta.py]: I have changed it to private method.

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Pull request merged**